sdk/python/kfp/components/_data_passing.py

```diff
@@ -0,0 +1,87 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = [
+    'type_to_type_name',
+    'type_name_to_type',
+    'type_to_deserializer',
+    'type_name_to_deserializer',
+    'type_name_to_serializer',
+]
+
+
+import inspect
+from typing import Any, Callable, NamedTuple, Sequence
+import warnings
+
+
+Converter = NamedTuple('Converter', [
+    ('types', Sequence[str]),
+    ('type_names', Sequence[str]),
+    ('serializer', Callable[[Any], str]),
+    ('deserializer_code', str),
+    ('definitions', str),
+])
+
+
+def _deserialize_bool(s) -> bool:
+    from distutils.util import strtobool
+    return strtobool(s) == 1
+
+
+_bool_deserializer_definitions = inspect.getsource(_deserialize_bool)
+_bool_deserializer_code = _deserialize_bool.__name__
+
+
+_converters = [
+    Converter([str], ['String', 'str'], str, 'str', None),
+    Converter([int], ['Integer', 'int'], str, 'int', None),
+    Converter([float], ['Float', 'float'], str, 'float', None),
+    Converter([bool], ['Boolean', 'bool'], str, _bool_deserializer_code, _bool_deserializer_definitions),
+]
+
+
+type_to_type_name = {typ: converter.type_names[0] for converter in _converters for typ in converter.types}
+type_name_to_type = {type_name: converter.types[0] for converter in _converters for type_name in converter.type_names if converter.types}
+type_to_deserializer = {typ: (converter.deserializer_code, converter.definitions) for converter in _converters for typ in converter.types}
+type_name_to_deserializer = {type_name: (converter.deserializer_code, converter.definitions) for converter in _converters for type_name in converter.type_names}
+type_name_to_serializer = {type_name: converter.serializer for converter in _converters for type_name in converter.type_names}
+
+
+def serialize_value(value, type_name: str) -> str:
+    '''serialize_value converts the passed value to string based on the serializer associated with the passed type_name'''
+    if type_name is None:
+        type_name = type_to_type_name.get(type(value), type(value).__name__)
+        warnings.warn('Missing type name was inferred as "{}" based on the value "{}".'.format(type_name, str(value)))
+
+    serializer = type_name_to_serializer.get(type_name, None)
+    if serializer:
+        return serializer(value)
+        #try:
+        #    return serializer(value)
+        #except Exception as e:
+        #    warnings.warn('Failed to serialize the value "{}" of type "{}" to type "{}". The value will be serialized using str(). Exception: {}'.format(
+        #        str(value),
+        #        str(type(value).__name__),
+        #        str(type_name),
+        #        str(e),
+        #    ))
```

Comment:
Should this be kept in the code base?

Reply:
I'm not yet fully sure what's the best way to handle serialization failures.

Reply:
Keeping it then makes sense.

Reply:
I've fixed this place. (BTW, it's part of the https://github.com/kubeflow/pipelines/pull/1934 PR)